experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c

```diff
@@ -338,6 +335,8 @@ static int ws_eink_init_display(struct ws_eink_fb_par *par)
 
 static int ws_eink_update_display(struct ws_eink_fb_par *par)
 {
+	ws_eink_reset(par);
```

Comment:
I don't think a hardware reset is the right solution here.  Take a look at the wiki: https://www.waveshare.com/wiki/2.13inch_e-Paper_HAT#Sleep_mode_.28Sleep.29

It says this:

> Sleep can make the module go into sleep mode to reduce the consumption.
>
> If you want to wake up the module from sleep mode, you need to give a LOW pulse to RST pin. Then maybe you need to reconfigure the parameter of power (According to the batches, some of them need to reconfigure, some needn’t). So if you want to wake up module, you had better use the Init function instead of Reset. Reset function and relative commands will be executed while executing the Init function. 

It's a bit hard to understand what they're saying, but I think they're saying that you should call `ws_eink_init_display()`

Reply:
Hi David

I sent mail to Waveshare supporter to clarify this  point and waiting his response.

Thanks
Thong Nguyen